### PR TITLE
INT: Update tinymce to version 7 and update dependencies.

### PR DESCRIPTION
**Pull request by kemister85**
Ticket: [INT-3310](https://ephocks.atlassian.net/browse/INT-3310)

Description:

* Add `license_key` config option.
* Update dependencies
* Updated to `"svelte": "^4.2.17"`.
* Update `tinymce` to `7.1.1`.
* Add missing Events.
* NOTE: Certain packages required React versions that are compatible with ^16.8.0, ^17.0.0, or ^18.0.0.

[INT-3310]: https://ephocks.atlassian.net/browse/INT-3310?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+- Update README.md to contain license key info.
+- Added storybook dependence's for various `storybook/...` packages.
+- Added `react` and `react-dom` to dev-dependencies due to certain packages required it.
+- Added events `Input`, `CompositionEnd`, `CompositionStart` & `CompositionUpdate`.
+- Added `licenseKey` to config option.
+
+### Changed
+
+- Bumped package version to `3.0.0-rc` from `2.0.3-rc`.
```
Comment (danoaky-tiny) [inline CHANGELOG.md:20]: Can leave this line out, as it's implied when it's no longer in the `## Unreleased` section

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Bumped `tinymce` version to `7.1.1` latest.
```
Comment (danoaky-tiny) [inline CHANGELOG.md:21]: - [ ] Also mention that the default cloud channel has been changed to 7 as well.

**Review by danoaky-tiny — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (kemister85) [reply CHANGELOG.md:20]:
```suggestion
```

File: CHANGELOG.md (hunk 2 above)
Comment (kemister85) [reply CHANGELOG.md:20]:
```suggestion
- Bumped `tinymce` version to `7.1.1` latest.
- Bumped default `cloud` channel to use `7`.
```

**Review by danoaky-tiny — APPROVED** [review]

File: src/main/component/Editor.svelte [hunk 3]
```diff
@@ -55,13 +61,15 @@
...
+  type Version = `${'4' | '5' | '6' | '7'}${'' | '-dev' | '-testing' | `.${number}` | `.${number}.${number}`}`;
 
   import { bindHandlers } from './Utils';
   export let id: string = uuid('tinymce-svelte'); // default values
   export let inline: boolean | undefined = undefined;
   export let disabled: boolean = false;
   export let apiKey: string = 'no-api-key';
-  export let channel: string = '6';
+  export let licenseKey: string | undefined = undefined;
+  export let channel: Version = '7';
```
Comment (jscasca) [inline src/main/component/Editor.svelte:72]: I feel like the Version type is a bit overkill. String should be good enough, and if we really want to keep the type maybe `Channel` would make more sense?

**Review by jscasca — APPROVED** [review]

File: src/main/component/Editor.svelte (hunk 3 above)
Comment (kemister85) [reply src/main/component/Editor.svelte:72]:
```suggestion
  export let channel: Channel = '7';
```

File: src/main/component/Editor.svelte [hunk 4]
```diff
@@ -55,13 +61,15 @@
   import { createEventDispatcher, onDestroy, onMount } from 'svelte';
   import type { TinyMCE, Editor as TinyMCEEditor } from 'tinymce';
   type EditorOptions = Parameters<TinyMCE['init']>[0];
+  type Version = `${'4' | '5' | '6' | '7'}${'' | '-dev' | '-testing' | `.${number}` | `.${number}.${number}`}`;
```
Comment (kemister85) [inline src/main/component/Editor.svelte:64]:
```suggestion
  type Channel = `${'4' | '5' | '6' | '7'}${'' | '-dev' | '-testing' | `.${number}` | `.${number}.${number}`}`;
```

**Pull request merged**